### PR TITLE
Floating point symfpu support 2

### PR DESCRIPTION
**Pull request by martin-cs**
More general improvements on the floating-point infrastructure in CVC4.
Includes the previous pull request ( https://github.com/CVC4/CVC4/pull/1103 ).

### CODE REVIEW

**Comment by 4tXJ7f** [comment]
Looks like fp/type_enumerator.h was missing in src/Makefile.am and that's why distcheck was failing. I added the missing line.

**Comment by martin-cs** [comment]
Thanks.  I think that may be a bug in my original branch.

File: src/theory/fp/type_enumerator.h [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+};/* FloatingPointEnumerator */
+
+class RoundingModeEnumerator : public TypeEnumeratorBase<RoundingModeEnumerator> {
+
+  RoundingMode rm;
+  bool enumerationComplete;
+
+public:
+
+  RoundingModeEnumerator(TypeNode type, TypeEnumeratorProperties * tep = NULL) throw(AssertionException) :
```
Comment (4tXJ7f) [inline src/theory/fp/type_enumerator.h]: Exception specification is [deprecated](http://en.cppreference.com/w/cpp/language/except_spec) in C++11.

File: src/theory/fp/theory_fp_type_rules.h [hunk 2]
```diff
@@ -464,8 +464,38 @@ class FloatingPointToRealTypeRule {
   }
 };
 
-} /* CVC4::theory::fp namespace */
-} /* CVC4::theory namespace */
-} /* CVC4 namespace */
+
+class CardinalityComputer {
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_type_rules.h]: The current draft of the code style guides for CVC4 [discourages](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#nonmember-static-member-and-global-functions) using classes with only static methods.

File: src/theory/fp/type_enumerator.h [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+ ** \verbatim
+ ** Original author: Morgan Deters
+ ** Major contributors: Martin Brain
+ ** Minor contributors (to current version): none
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2015  New York University and The University of Iowa
+ ** See the file COPYING in the top-level source directory for licensing
+ ** information.\endverbatim
+ **
+ ** \brief An enumerator for floating-point numbers.
```
Comment (4tXJ7f) [inline src/theory/fp/type_enumerator.h]: ... and rounding modes?

Comment (mpreiner) [reply src/theory/fp/type_enumerator.h]: Seems like this file still has the old header format. Need to check if there are more of them.

Comment (4tXJ7f) [reply src/theory/fp/type_enumerator.h]: Yeah, it looks like this is also the case in the newest version of the branch.

**Pull request merged**